### PR TITLE
Enhance Docstrings for Scheduler Classes in Mesa Time Module

### PR DESCRIPTION
**Pull request by EwoutH**
This commit improves the docstrings for each scheduler class and some key methods in the Mesa Time Module. The enhancements include detailed descriptions of the classes and methods, clarifying their functionality, usage, and the expected behavior of each scheduler type. 

Improvements include:
- Added comprehensive class descriptions for `BaseScheduler`, `RandomActivation`, `SimultaneousActivation`, `StagedActivation`, and `RandomActivationByType`, highlighting their specific roles and behaviors in agent activation within simulations.
- Included detailed method descriptions for `add`, `remove`, `step`, and other relevant methods, explaining their functionality and the parameters they use.
- Specified attributes for each class, providing clarity on the data each scheduler maintains and manipulates.
- Enhanced readability and understanding for users who need to implement or modify schedulers in their agent-based models.

No code or functional changes, only documentation.

### CODE REVIEW

File: mesa/time.py [hunk 1]
```diff
@@ -39,12 +39,24 @@
...
-    Assumes that each agent added has a *step* method which takes no arguments.
-
-    (This is explicitly meant to replicate the scheduler in MASON).
+    """
+    A simple scheduler that activates agents one at a time, in the order they were added.
+
+    This scheduler is designed to replicate the behavior of the scheduler in MASON, a multi-agent simulation toolkit.
+    It assumes that each agent added has a `step` method which takes no arguments and executes the agent's actions.
+
+    Attributes:
```
Comment (rht) [inline mesa/time.py:48]: It currently renders without formatting: https://mesa--1873.org.readthedocs.build/en/1873/apis/time.html.

It seems that you could use info fields: https://stackoverflow.com/questions/8649105/how-to-show-instance-attributes-in-sphinx-doc, which renders better. This is not a blocker to this PR getting merged. Only if you have the time for it.

File: mesa/time.py [hunk 2]
```diff
@@ -39,12 +39,24 @@
...
+        model (Model): The model instance associated with the scheduler.
+        steps (int): The number of steps the scheduler has taken.
+        time (TimeT): The current time in the simulation. Can be an integer or a float.
+        _agents (dict[int, Agent]): A private dictionary mapping agent IDs to agent instances.
```
Comment (rht) [inline mesa/time.py:52]: This shouldn't be documented. The private implementation detail may change, while the API (`BaseScheduler.agents`) doesn't.

File: mesa/time.py [hunk 3]
```diff
@@ -131,11 +149,20 @@ def step(self) -> None:
...
 class SimultaneousActivation(BaseScheduler):
-    """A scheduler to simulate the simultaneous activation of all the agents.
+    """
+    A scheduler that simulates the simultaneous activation of all agents.
+
+    This scheduler is unique in that it requires agents to have both `step` and `advance` methods.
+    The `step` method is for activating the agent and staging any changes without applying them immediately.
+    The `advance` method then applies these changes, simulating simultaneous action.
+
+    This scheduler is useful in scenarios where the interactions between agents are sensitive to the order
```
Comment (rht) [inline mesa/time.py:159]: This paragraph is informative.

File: mesa/time.py [hunk 4]
```diff
@@ -150,15 +177,25 @@ def step(self) -> None:
...
-    """A scheduler which allows agent activation to be divided into several
-    stages instead of a single `step` method. All agents execute one stage
+    """
+    A scheduler allowing agent activation to be divided into several stages, with all agents executing one stage
     before moving on to the next.
 
-    Agents must have all the stage methods implemented. Stage methods take a
-    model object as their only argument.
+    This scheduler is useful for complex models where actions need to be broken down into distinct phases
+    for each agent in each time step. Agents must implement methods for each defined stage.
```
Comment (rht) [inline mesa/time.py:185]: May add that this class is a generalization of `SimultaneousActivation`.

File: mesa/time.py (hunk 2 above)
Comment (EwoutH) [reply mesa/time.py:52]: `model.schedule.steps` (or `self.schedule.steps`) is frequently used to get the current timestep. I think this should be documented somewhere. Or is there another best practice to get the current timestep?

File: mesa/time.py (hunk 1 above)
Comment (EwoutH) [reply mesa/time.py:48]: I just added list markers (`- `), that seems to do the trick

File: mesa/time.py (hunk 2 above)
Comment (rht) [reply mesa/time.py]: Just `_agents`. Everything else is fine.

Comment (EwoutH) [reply mesa/time.py]: I removed `_agents` from the docs.

File: mesa/time.py (hunk 4 above)
Comment (EwoutH) [reply mesa/time.py:184]: Done!

**Review by rht — APPROVED** [review]

**Comment by rht** [comment]
Ruff and Black CI failed.

**Comment by EwoutH** [comment]
Fixed

**Comment by rht** [comment]
Merged, thank you.

**Pull request merged**